script/cms/compare-json.js

```diff
@@ -0,0 +1,322 @@
+  });
+
+  // Get array of differences. Exclude new properties because transformed
+  // CMS objects may have additional properties
+  return deepDiff(graphQlObject, cmsExportObject)
+    ?.filter(d => d.kind !== 'N')
+    .map(diff => {
+      return {
+        diffType: getDiffType(diff.kind),
+        path: diff.index
+          ? `${diff.path.join('/')}[${diff.index}]`
+          : diff.path.join('/'),
+        ...(diff.item && { item: getDiffItem(diff.item) }),
+        ...(diff.lhs && { graphQL: diff.lhs }),
+        ...(diff.rhs && { cmsExport: diff.rhs }),
+      };
+    });
+};
+
+/**
+ * Handles running and outputting the diff based on the command line input
+ */
+const runComparison = () => {
+  if (entityNames) {
+    const rawEntities = [];
+    const transformedEntities = entityNames.map(entityName => {
+      const nodeNamePieces = entityName.split('.').slice(0, 2);
+      assert(
+        nodeNamePieces.length === 2,
+        '--entity (or -e) needs to be the filename of an entity. E.g. node.<uuid>.json',
+      );
+
+      const unTransformedEntity = readEntity(exportDir, ...nodeNamePieces);
+      rawEntities.push(unTransformedEntity);
+
+      return assembleEntityTree(unTransformedEntity, false);
+    });
+
+    let graphQlObject;
+
+    // Handle paragraph entities
+    if (rawEntities[0].baseType !== 'node') {
+      let fileNames = fs
+        .readdirSync(exportDir)
+        .filter(fn => fn !== 'meta')
+        .map(name => name.split('.').slice(0, 2));
+
+      fileNames = fileNames.filter(
+        ([baseType]) => baseType === 'node' || baseType === 'paragraph',
+      );
+
+      const cmsEntities = map(fileNames, entityDetails =>
+        readEntity(exportDir, ...entityDetails),
+      );
+
+      // Find parent node for paragraph object
+      const parentNode = getParentNode(rawEntities[0], cmsEntities);
+      const graphQlNode = graphQL.data.nodeQuery.entities.filter(
+        e =>
+          e.entityBundle === parentNode.node.entityBundle &&
+          parseInt(e.entityId, 10) === parentNode.node.nid[0].value,
+      )[0];
+
+      // Get specific paragraph object
+      graphQlObject = graphQlNode[camelCase(parentNode.path)][0];
+    } else {
+      // Find GraphQL object from pages.json data
+      graphQlObject = graphQL.data.nodeQuery.entities.filter(
+        e =>
+          e.entityBundle === rawEntities[0].entityBundle &&
+          parseInt(e.entityId, 10) === rawEntities[0].nid[0].value,
+      )[0];
+    }
+
+    if (graphQlObject) {
+      const diff = compareJson(graphQlObject, transformedEntities[0]);
+      if (diff) {
+        fs.writeFileSync(
+          path.join(__dirname, `../../content-object-diff.json`),
+          JSON.stringify(diff),
+        );
+        console.log(`${diff.length} differences: './content-object-diff.json'`);
+      } else {
+        console.log(`No differences found!`);
+      }
+    } else {
+      console.log('Node not present in .cache/localhost/drupal/pages.json');
```

Comment:
Will this always be /localhost/ ?

Reply:
Nice catch, this won't always be localhost. Forgot to update this after working on this locally.

Reply:
I updated this in a previous commit